eZ/Publish/Core/Repository/LocationService.php

```diff
@@ -350,7 +350,7 @@ public function createLocation(ContentInfo $contentInfo, LocationCreateStruct $l
         $content = $this->repository->getContentService()->loadContent($contentInfo->id);
         $parentLocation = $this->loadLocation($locationCreateStruct->parentLocationId);
 
-        if (!$this->repository->canUser('content', 'manage_locations', $content->contentInfo, $parentLocation)) {
+        if (!$this->repository->canUser('content', 'manage_locations', $content->contentInfo, $content)) {
```

Comment:
hmm, this change does not make locally sense atm to me.
1. I don't think context can be a content object
2. even if it did work, it would probably pick main location, which is not the ocation we are clearing a location in here.


See what is going on in SubtreeLimitaiton:
```php
        foreach ($targets as $target) {
            if (!$target instanceof Location && !$target instanceof SPILocation) {
                // As this is Role limitation we need to signal abstain on unsupported $targets
                return self::ACCESS_ABSTAIN;
            }

            foreach ($value->limitationValues as $limitationPathString) {
                if ($target->pathString === $limitationPathString) {
                    return true;
                }
                if (strpos($target->pathString, $limitationPathString) === 0) {
                    return true;
                }
            }
        }
```

So if given a content, it will return `ACCESS_ABSTAIN`, on role limitation that will allow it _(assign role with subtree limitation)_, and for policy limitation it will *not*. If you want to look, see PermissionResolver:
```php
                    if ($accessVote !== LimitationType::ACCESS_GRANTED) {
                        $limitationsPass = false;
                        break; // Break to next policy, all limitations must pass
                    }
```

Reply:
@kmadejski I think what you rather want to do is like in legacy, check if user has access to manage_locations **OR** create.

see: https://github.com/ezsystems/ezpublish-legacy/blob/d76ce9824dabaceeefa77530c8c611e447d1b109/design/admin/templates/locations.tpl#L7

Will that still solve the original CS issue?

Reply:
@andrerom thanks for the review. Let me explain the issue.

*Content structure*:
- `/Root/ (/1/2/)` (folder)
- `/Root/Free` (folder)
- `/Root/Restricted` (folder)
- `/Root/Restricted/Not-shareable` (article)
- `/Root/Restricted/Available-for-share` (folder)
- `/Root/Restricted/Available-for-share/Shareable` (article)

*Roles*:
- Contributors

*Policies (for `Contributors` role)*:
- `content/create` limited on: 
-- subtrees: `/Root/Free` and `/Root/Restricted/Available-for-share`
-- sections: `Media` and `Standard`
- `content/manage_locations` limited on:
-- subtrees: `/Root/Free` and `/Root/Restricted/Available-for-share`
-- sections: `Media` and `Standard`
- additional policies required to be able to log-in to BO, read content, etc. 

*Problem*:
The contributor should be able to add a new location of all contents under `/Root/Restricted/Available-for-share` subtree but not for other contents under `/Root/Restricted` subtree. The new location could be located only under `/Root/Free` subtree.

*Cases*:
1. The contributor tries to add the new location of `/Root/Restricted/Not-shareable` under `/Root/Free` subtree.
2. The contributor tries to add the new location of `/Root/Restricted/Available-for-share/Shareable` under `/Root/Free` subtree.

*Legacy (5.4) behavior*:
Case 1: The contributor **can't** add the new location of `/Root/Restricted/Not-shareable` under `/Root/Free` subtree. The **Add locations** button is disabled.
Case 2: The contributor **can** add the new location of `/Root/Restricted/Available-for-share/Shareable` under `/Root/Free` subtree.

*eZ Platform (1.13) behavior*:
Case 1: The contributor **can** add the new location of `/Root/Restricted/Not-shareable` under `/Root/Free` subtree. ❌ 
Case 2: The contributor **can** add the new location of `/Root/Restricted/Available-for-share/Shareable` under `/Root/Free` subtree. 

*eZ Platform (including https://github.com/ezsystems/ezpublish-kernel/pull/2283/commits/f76dd81db20b485f1922032f4f569df6bf98f991) behavior*:
Case 1: The contributor **can't** add the new location of `/Root/Restricted/Not-shareable` under `/Root/Free` subtree. The contributor gets an exception (red bar at the bottom of page). 
Case 2: The contributor **can** add the new location of `/Root/Restricted/Available-for-share/Shareable` under `/Root/Free` subtree.